src/Datadog.Trace/Configuration/ConfigurationSourceExtensions.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Configuration
+{
+    internal static class ConfigurationSourceExtensions
+    {
+        private const char Splitter = ';';
+
+        public static IEnumerable<string> GetStrings(this IConfigurationSource source, string key)
+        {
+            return source?.GetString(key)
+                ?.Split(new[] { Splitter }, StringSplitOptions.RemoveEmptyEntries)
+                ?? Enumerable.Empty<string>();
+        }
+
+        public static TEnum GetTypedValue<TEnum>(this IConfigurationSource source, string key)
+            where TEnum : struct, IConvertible
+        {
+            Enum.TryParse(source?.GetString(key) ?? "default", ignoreCase: true, out TEnum typedValue);
+            return typedValue;
+        }
+
+        public static HashSet<TEnum> GetTypedValues<TEnum>(this IConfigurationSource source, string key)
```

Comment:
Why returning a `HashSet` here? It makes the code not "orthogonal" as e.g. `GetStrings` returns `IEnumerable`.
the caller can always convert to HashSet. 

Also I do not like the undocumented side-effect of setting the `default`. Especially that the default depends on the context.
See:
> Platforms such as ASP.NET may pre-configure out-of-the-box propagators. If pre-configured, Propagators SHOULD default to a composite Propagator containing the W3C Trace Context Propagator and the Baggage Propagator specified in the Baggage API. These platforms MUST also allow pre-configured propagators to be disabled or overridden.

From: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#global-propagators

Reply:
agree, seems like design error.